[Andrew Lo] iP

wrewsama opened the pull request:
# This is a Heading

* This
* Is a 
* Bullet
* List

1. This
2. Is a
3. Numbered
4. List

```java
public static void main(String[] args) {
    System.out.println("Here's a code block.");
}
```

- [ ] task
- [ ] list

emoji check :+1:

>Blockquote

[Link](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)

Some `inline.code`

**Some** _text_ ~~formatting~~

Commit:
Add Gradle support

wrewsama committed:
Implemented a skeletal version of Duke that starts by greeting the user, simply echos commands entered by the user, and exits when the user types bye.

wrewsama committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

wrewsama committed:
Add Task class to encapsulate a user-added task

wrewsama committed:
Add marking and unmarking functionality

wrewsama committed:
Add Deadline class and input functionality

wrewsama committed:
Add Todo class and functionality

wrewsama committed:
Add event class and functionality

wrewsama committed:
Set up I/O Redirection Text UI Testing

wrewsama committed:
Add exception class and error handling

wrewsama committed:
Add delete functionality

wrewsama committed:
Add error handling to addDeadline()

wrewsama committed:
Refactor addTodo's error handling

wrewsama committed:
Add error handling to addEvent()

wrewsama committed:
Add error handling to mark, unmark, and delete functions

wrewsama committed:
Refactor commands to use Enums

wrewsama committed:
Add saving and loading to local storage

wrewsama committed:
Change Deadline and Event to store dates as LocalDate objects

wrewsama committed:
Add function to print all tasks on a selected date

wrewsama committed:
Merge branch 'branch-Level-7'

wrewsama committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Task.java

wrewsama committed:
Resolve merge conflicts fix bugs

wrewsama committed:
Rename filehandler to Storage

wrewsama committed:
Refactor functions into Ui, Storage, and Parser classes

wrewsama committed:
Extract functionality into separate classes for UI, Storage, Parser, and Task List

wrewsama committed:
Clean up unused code

wrewsama committed:
Move classes into duke package

wrewsama committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

wrewsama committed:
Set up Gradle

wrewsama committed:
Add unit tests

wrewsama committed:
Fix bugs and build JAR

wrewsama committed:
Write javadocs

wrewsama committed:
Set up checkstyle config

wrewsama committed:
Clean up javadocs

wrewsama committed:
Refactor in accordance with Java Coding Standard

wrewsama committed:
Add find function

wrewsama committed:
Merge tag 'A-JavaDoc'

wrewsama committed:
Merge branch 'branch-A-CodingStandard'

wrewsama committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

wrewsama committed:
Improve style

wrewsama committed:
Merge branch 'A-CheckStyle'

wrewsama committed:
Set up GUI to echo user inputs

wrewsama committed:
Complete GUI with user input but with dummy function to get output

wrewsama committed:
Move all IO from CLI to GUI

wrewsama committed:
Update javadocs

wrewsama committed:
Merge branch 'Level-10'

wrewsama committed:
Add varargs

misterpuffin commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,47 @@
+     * @return corresponding command if the name is valid
+     *         null otherwise
+     */
+    public static Command get(String name) {
```
not a coding standard violation, but was thinking that the Enum.valueOf() function would be useful here!

wrewsama replied:
Thanks for the tip!

misterpuffin commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @param description the description of the task
+     * @param by the due date of the task
+     */
```
good to add @throws in javadoc

wrewsama replied:
Good catch, thanks!

misterpuffin commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,45 @@
+package duke;
```
minor nit but might want to add new line here for consistency across the files

wrewsama replied:
Nice catch, I completely did not notice this.

misterpuffin commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,31 @@
+     */
+    public DukeException(String message) {
+        super("OOPS!!!" + message);
+        this.message = message;
```
just a suggestion but perhaps you don't have to save the message in a field since you're calling super with the new message already! then you won't have to override toString() method.

misterpuffin commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,284 @@
+    }
+
+    /**
+     * Remove a task from the task list.
```
```suggestion
     * Removes a task from the task list.
```

misterpuffin commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,51 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DukeTest {
+
+
+
```
Unnecessary blank lines

misterpuffin commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,51 @@
+        } catch (DukeException e) {
+            System.out.println(e);
+        }
+
```
same here

misterpuffin reviewed (commented):
Overall, really good work! The code was very readable and I had trouble finding code standard violations. All the best for the remainder of your iP!

jsincorporated commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,47 @@
+ * @author Andrew Lo Zhi Sheng
+ * @version CS2103T AY22/23 Semester 1
+ */
+public enum Command {
```
Good use of enums!

wrewsama replied:
Thanks!

jsincorporated commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,103 @@
+
+        // Determine the action to perform
+        switch (command) {
+        case BYE:
```
Nice job following the coding standards✅

wrewsama replied:
Thanks!

jsincorporated reviewed (commented):
Overall, in terms of code quality, I found that it was pretty well done. There are just some individual lines where I feel breaking up the logic would help in readability and understanding of the code. Other than that, I feel that the code quality is really quite solid. Great job!

wrewsama reviewed (commented):
Thanks for the comments!

wrewsama committed:
Fix style

wrewsama committed:
Add assertions

wrewsama committed:
Duke::start initialises and sets up various JavaFx objects

The current implementation of the method is too long (>30 LoC) and has more than a single level of abstraction.

Let's abstract out similar functionalities into private helper methods enables us to achieve a single level of abstraction and greatly shorten the length of the code within the method.

This will enhance the overall readability of the codebase.

wrewsama committed:
Rename variables

wrewsama committed:
TaskList::find uses a for-each loop to go through each task to find descriptions matching a given key.

This is not a very elegant solution.

Let's use stream operations instead.

This will make the codebase cleaner and more elegant.

wrewsama committed:
Merge pull request #1 from wrewsama/A-Assertions

Add assertions

wrewsama committed:
Merge branch 'master' of https://github.com/wrewsama/ip

wrewsama committed:
Merge branch 'master' into A-CodeQuality

wrewsama committed:
Merge branch 'master' into A-Streams

wrewsama committed:
Merge pull request #3 from wrewsama/A-CodeQuality

A code quality

wrewsama committed:
Merge branch 'master' into A-Streams

wrewsama committed:
Merge pull request #2 from wrewsama/A-Streams

A-Streams

wrewsama committed:
Set up Continuous Integration

wrewsama committed:
Add automatic sorting

wrewsama committed:
Merge branch 'BCD-Ext'

wrewsama committed:
Rename branch and PR

wrewsama committed:
Rename branch and PR

wrewsama committed:
Rename branch and PR

wrewsama committed:
Merge pull request #4 from wrewsama/branch-A-Assertions

Rename branch and PR

wrewsama committed:
Merge branch 'master' of https://github.com/wrewsama/ip

wrewsama committed:
Merge branch 'master' into branch-A-CodeQuality

wrewsama committed:
Merge branch 'master' into branch-A-Streams

wrewsama committed:
Merge pull request #5 from wrewsama/branch-A-CodeQuality

Rename branch and PR

wrewsama committed:
Merge branch 'master' of https://github.com/wrewsama/ip

wrewsama committed:
Merge branch 'master' into branch-A-Streams

wrewsama committed:
Merge pull request #6 from wrewsama/branch-A-Streams

Rename branch and PR

wrewsama committed:
Merge branch 'master' of https://github.com/wrewsama/ip

wrewsama committed:
Change UI font

wrewsama committed:
Change duke responses

wrewsama committed:
Add more error handling

wrewsama committed:
Add test for task::unmark

wrewsama committed:
Add Ui screenshot

wrewsama committed:
Update User Guide

wrewsama committed:
Fix style and javadocs

wrewsama committed:
Fix jar file creation

wrewsama committed:
Fix style in test

wrewsama committed:
Fix javadocs
